src/guest_memory.rs

```diff
@@ -1179,19 +1321,24 @@ mod tests {
         assert!(mem.write_obj(obj, addr).is_ok());
         assert!(mem.read_obj::<ZeroSizedStruct>(addr).is_ok());
 
-        assert_eq!(mem.read_from(addr, &mut Cursor::new(&image), 0).unwrap(), 0);
+        assert_eq!(
+            mem.read_volatile_from(addr, &mut image.as_slice(), 0)
+                .unwrap(),
+            0
+        );
 
         assert!(mem
-            .read_exact_from(addr, &mut Cursor::new(&image), 0)
+            .read_exact_volatile_from(addr, &mut image.as_slice(), 0)
             .is_ok());
```

Comment:
Nit: It is more rigorous to match results exactly (also applies to all other cases where you've edited the assertion)
```suggestion
        asser_eq!(mem
            .read_exact_from(addr, &mut Cursor::new(&image), 0)
            .read_exact_volatile_from(addr, &mut image.as_slice(), 0),
            Ok(0)
        );
```